test/exmg-form.spec.ts

```diff
@@ -0,0 +1,68 @@
+import {ExmgForm} from '../exmg-form';
+import {promisifyFlush, onExmgFormCancel, onExmgFormSubmit} from './utils';
+import {PaperInputElement} from "@polymer/paper-input/paper-input";
+
+declare const fixture: <T extends HTMLElement = HTMLElement>(id: string, model?: object) => T;
+declare const flush: (cb?: Function) => void;
+
+const {assert} = chai;
+
+suite('<exmg-form>', function () {
+  let element: ExmgForm;
+  const flushCompleted = promisifyFlush(flush);
+
+  suite('base usage', function () {
+    setup(() => {
+      element = fixture('ExmgFormBasicElement');
+    });
+
+    test('element is upgraded', function () {
+      assert.instanceOf(element, ExmgForm);
+    });
+
+    test('submitting form', async () => {
+      await flushCompleted();
+
+      const field1Input = <PaperInputElement>element.querySelector('paper-input[name=field1]');
```

Comment:
Function querySelector accept generic type
`querySelector<E extends Element = Element>(selectors: string): E | null;`
instead casting you write it in proper form
`element.querySelector<PaperInputElement>('paper-input[name=field1]');`